core/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -759,15 +830,16 @@ class DeltaAnalysis(session: SparkSession)
           Cast(input, dt, Option(timeZone), ansiEnabled = false)
       case SQLConf.StoreAssignmentPolicy.ANSI =>
         (input: Expression, dt: DataType, name: String) => {
-          AnsiCast(input, dt, Option(timeZone))
+          val cast = Cast(input, dt, Option(timeZone), ansiEnabled = true)
```

Comment:
What is this change for?

Reply:
Compile error; `AnsiCast` removed in 3.4 and consolidated into one `Cast` class